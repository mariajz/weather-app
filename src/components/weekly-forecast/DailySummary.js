import React from 'react';
import { View as WeatherIconWrapper } from 'react-native';
import {
    ChancesOfRain,
    DailySummaryWrapper,
    DayText,
    MaxTemperature,
    MinTemperature,
} from './WeeklyForecast.style';
import { Divider } from '../../commons/styles';
import { degreeSymbol } from '../../constants/Constants';
import { CustomImage } from '../../commons/visual-elements';

const DailySummary = ({ dailyWeatherItem }) => {
    const { maxTemp, minTemp, willItRain, chanceOfRain, day, condition } =
        dailyWeatherItem;

    return (
        <>
            <Divider />
            <DailySummaryWrapper>
                <DayText>{day}</DayText>
                <WeatherIconWrapper>
                    <CustomImage path={condition} width={20} height={20} />
                    <If condition={willItRain}>
                        <ChancesOfRain>{chanceOfRain}%</ChancesOfRain>
                    </If>
                </WeatherIconWrapper>
                <MinTemperature>
                    {minTemp}
                    {degreeSymbol}C
                </MinTemperature>
                <MaxTemperature>
                    {maxTemp}
                    {degreeSymbol}C
                </MaxTemperature>
            </DailySummaryWrapper>
        </>
    );
};

export default DailySummary;
