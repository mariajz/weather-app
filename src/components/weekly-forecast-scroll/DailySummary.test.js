import { render } from '@testing-library/react-native';
import React from 'react';
import DailySummary from './DailySummary';

const defaultProps = {
    dailyWeatherItem: {
        chanceOfRain: 0,
        condition: 10,
        day: '2023-08-27',
        maxTemp: 25.6,
        minTemp: 20.2,
        willItRain: 0,
    },
};
describe('DailySummary', () => {
    it('should render DailySummary', () => {
        const container = render(<DailySummary {...defaultProps} />);

        expect(container).toMatchSnapshot();
    });
});
