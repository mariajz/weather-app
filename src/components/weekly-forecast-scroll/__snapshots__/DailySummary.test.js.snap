// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailySummary should render DailySummary 1`] = `
[
  <View
    style={
      {
        "borderColor": "grey",
        "borderStyle": "solid",
        "borderWidth": 0.5,
        "marginTop": 5,
        "width": "100%",
      }
    }
  />,
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "space-between",
        "paddingBottom": 6,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 6,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 12,
          "marginLeft": 0,
        }
      }
    >
      2023-08-27
    </Text>
    <View>
      <Image
        height={20}
        source={10}
        style={
          {
            "height": 20,
            "width": 20,
          }
        }
        testID="image"
        width={20}
      />
    </View>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 12,
          "marginLeft": 0,
        }
      }
    >
      20.2
      °
      C
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 12,
          "marginLeft": 0,
        }
      }
    >
      25.6
      °
      C
    </Text>
  </View>,
]
`;
